### PR TITLE
Deprecate and rename findFirstObject and getObjectIDPosition

### LINKED ISSUE #101: Implement two new methods for unit testing of relevance
- [Specs and CTS tests](https://github.com/algolia/algoliasearch-client-specs-internal/pull/41)
- [Go client base implementation](https://github.com/algolia/algoliasearch-client-go/pull/524)

### PR DESCRIPTION
**Pull request by q-litzler**
fixes #101

### CODE REVIEW

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt [hunk 1]
```diff
@@ -122,25 +122,37 @@ public interface EndpointSearch {
         requestOptions: RequestOptions? = null
     ): ResponseSearch
 
+    @Deprecated(
+        message = "Use findObject instead.",
+        replaceWith = ReplaceWith("findObject(match, query, doNotPaginate, requestOptions)"),
```
Comment (nunomaduro) [inline src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]:
```suggestion
        replaceWith = ReplaceWith("findObject(match, query, paginate, requestOptions)"),
```

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt [hunk 2]
```diff
@@ -122,25 +122,37 @@ public interface EndpointSearch {
...
+        level = DeprecationLevel.ERROR
```
Comment (nunomaduro) [inline src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]: It's not an warning?

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearchImpl.kt [hunk 3]
```diff
@@ -35,20 +35,29 @@ internal class EndpointSearchImpl(
...
     }
 
-    override tailrec suspend fun findFirstObject(
+    override suspend fun findFirstObject(
         match: (ResponseSearch.Hit) -> Boolean,
         query: Query,
         doNotPaginate: Boolean,
         requestOptions: RequestOptions?
+    ): ResponseHitWithPosition? {
+        return findObject(match, query, doNotPaginate, requestOptions)
```
Comment (nunomaduro) [inline src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearchImpl.kt]: You don't need to inverse `doNotPaginate` to `! doNotPaginate` ?

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt (hunk 1 above)
Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]: Excellent catch 💯

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearchImpl.kt (hunk 3 above)
Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearchImpl.kt]: you make some good points 👍

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt (hunk 1 above)
Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]: Spoke too fast, actually, it need stay as `doNotPaginate`, this allows the IDE to replace the old variable with the new one (So it needs to have the name of the old variable to perform the match properly)

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt (hunk 2 above)
Comment (q-litzler) [reply src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]: `ERROR` will force developers to change it asap, as it prevents compilation. 

Maybe it's too harsh, I will lower it to `WARNING`

**Pull request merged**